### ISSUE TITLE
bintray/jcenter will stop serving packages on February 1st 2022

### ISSUE THREAD

**Issue by jcesarmobile** [ionic-team/capacitor#4266]
The cordova-android dependency is fetched from there, so apps won't be able to fetch the dependency when that moment comes.
We should find some alternative way of fetching cordova-android

**Comment by rt012** [comment]
Any news on this?

**Comment by beruic** [comment]
Given the time of year, this seems more and more like a high priority issue.
@rt012 can you elaborate a bit more on exactly what you did?

I have only tried replacing `jcenter()` usage in `android/build.gradle` with `mavenCentral()`, but that just gives an error that `org.apache.cordova:framework:7.0.0` cannot be found. This also seems like quite an old version, so I guess chances are slim of making the Cordova people upload it to Maven Central. Thoughts?

**Comment by beruic** [comment]
Seems like the version in quest is available from [Spring Plugins](https://mvnrepository.com/artifact/org.apache.cordova/framework?repo=springio-plugins-release), so there might be a quick fix available. That said, there may be other failing dependencies, and I guess it still needs to be fixed more permanently.

**Comment by jcesarmobile** [comment]
JFrog said they are going to keep serving packages from jcenter indefinitely

**Comment by beruic** [comment]
> JFrog said they are going to keep serving packages from jcenter indefinitely

@jcesarmobile  Do you have a link for this?

**Comment by jcesarmobile** [comment]
The original article from jfrog was updated:

> UPDATE 4/27/2021: We listened to the community and will keep JCenter as a read-only repository indefinitely. Our customers and the community can continue to rely on JCenter as a reliable mirror for Java packages.



https://jfrog.com/blog/into-the-sunset-bintray-jcenter-gocenter-and-chartcenter/

**Comment by brentspector** [comment]
I also had this issue.

> Given the time of year, this seems more and more like a high priority issue. @rt012 can you elaborate a bit more on exactly what you did?
> 
> I have only tried replacing `jcenter()` usage in `android/build.gradle` with `mavenCentral()`, but that just gives an error that `org.apache.cordova:framework:7.0.0` cannot be found. This also seems like quite an old version, so I guess chances are slim of making the Cordova people upload it to Maven Central. Thoughts?

I had the same issue. I ended up changing the `cordovaAndroidVersion` to `10.1.1` in `variables.gradle` in order to pull a newer version from mavenCentral. I also updated `androidxAppCompatVersion` to `1.4.0` , `junitVersion` to `4.13.2` , `androidxJunitVersion` to `1.1.3` and `androidxEspressoCoreVersion` to `3.4.0`. They all worked with my app (which is admittedly a pretty simple app). 

Also I replaced `jcenter()` with `mavenCentral()`. I tried to add a custom maven repository pointed to `repo.spring.io` where `7.0.0` is hosted, but they closed anonymous requests down. https://spring.io/blog/2020/10/29/notice-of-permissions-changes-to-repo-spring-io-fall-and-winter-2020

**Comment by allanwest** [comment]
Gradle has been warning about this since February 2021, as well: 
https://blog.gradle.org/jcenter-shutdown
As well as Android Developer Docs
https://developer.android.com/studio/releases/gradle-plugin

Seeing as both Android Grade Plugin (AGP) and Gradle version have been aligned over the course of 2021, maybe it would also make sense to get the Gradle version bumped up to 7.X.X, latest version is currently 7.3.3.

**Comment by hoi4** [comment]
When using cordova plugins like e.g. `cordova-plugin-screen-orientation`, using `cordova-android` `10.1.1` from `mavenCentral()` does not work. 

I get the following error in `CDVOrientation.java`
```
75: Activity activity = cordova.getActivity();
                                 ^
  class file for androidx.appcompat.app.AppCompatActivity not found
```

**Comment by johannds** [comment]
This is an issue that will cause build outages for every single project that uses Capacitor after the end of January 2022.

What is the timeline for addressing this issue?

**Comment by jcesarmobile** [comment]
No, read the comments, they are going to keep serving the packages indefinitely.

jCenter will be removed in Capacitor 4, this can't be done in Capacitor 3 because it's a breaking change. Just added it to the 4.0.0 milestone a few hours before you commented.
